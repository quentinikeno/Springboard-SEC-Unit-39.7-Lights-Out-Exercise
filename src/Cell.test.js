import React from "react";
import { render } from "@testing-library/react";
import Cell from "./Cell";

describe("<Cell/>", () => {
	let container;

	beforeEach(function () {
		let tr = document.createElement("tr");
		container = document.body.appendChild(tr);
	});

	test("if it renders without crashing", () => {
		render(<Cell />, { container });
	});

	test("that it matches the snapshot when lit", () => {
		const { asFragment } = render(<Cell isLit={true} />, { container });
		expect(asFragment()).toMatchSnapshot();
	});

	test("that it matches the snapshot when not lit", () => {
		const { asFragment } = render(<Cell isLit={false} />, { container });
		expect(asFragment()).toMatchSnapshot();
	});
});
