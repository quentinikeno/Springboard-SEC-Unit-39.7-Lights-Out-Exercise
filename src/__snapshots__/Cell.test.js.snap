// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell/> that it matches the snapshot when lit 1`] = `<DocumentFragment />`;

exports[`<Cell/> that it matches the snapshot when not lit 1`] = `<DocumentFragment />`;
